addons/mail/models/mail_message.py

```diff
@@ -582,6 +593,9 @@ def create(self, values_list):
             if 'record_name' not in values and 'default_record_name' not in self.env.context:
                 values['record_name'] = self._get_record_name(values)
 
+            if 'is_internal' not in values and 'default_is_internal' not in self.env.context and values.get('subtype_id'):
+                values['is_internal'] = self.env['mail.message.subtype'].browse(values['subtype_id']).internal
```

Comment:
What is the point of assigning an `is_internal` value that matches the one from the subtype if it does not override it in security check? It could always be set to default value False and it would work all the same. (same behavior as the one you explained will happen for old messages)

Reply:
Finally I removed completely the synchronization. Indeed migration could be complicated and backward support would lead to too much code. Internal is now working like the old flag.